app/celery/process_ses_receipts_tasks.py

```diff
@@ -205,6 +205,9 @@ def process_ses_results(  # noqa: C901 (too complex 14 > 10)
             )
             return
 
+        if incoming_status in (NOTIFICATION_PERMANENT_FAILURE, NOTIFICATION_DELIVERED):
```

Comment:
Are these all the final states?  What about NOTIFICATION_CANCELLED and the other elements in constants.py::NOTIFICATION_STATUS_TYPES_COMPLETED?